crypto/src/keys.rs

```diff
@@ -150,11 +190,34 @@ impl Diversifier {
     }
 
     #[allow(non_snake_case)]
-    pub fn derive_transmission_key(&self, ivk: IncomingViewingKey) -> TransmissionKey {
+    pub fn derive_transmission_key(&self, ivk: &IncomingViewingKey) -> TransmissionKey {
```

Comment:
should this be a method on the `Diversifier` or on the `IncomingViewingKey`? since there are multiple diversifiers for the same IVK, maybe it would make more sense for it to be a method on the IVK rather than the diversifier?